packages/http-client-python/generator/pygen/codegen/serializers/__init__.py

```diff
@@ -122,9 +135,13 @@ def serialize(self) -> None:
         for client_namespace, client_namespace_type in self.code_model.client_namespace_types.items():
             exec_path = self.exec_path(client_namespace)
             if client_namespace == "":
-                # Write the setup file
-                if self.code_model.options["basic-setup-py"]:
-                    self.write_file(exec_path / Path("setup.py"), general_serializer.serialize_setup_file())
+                # remove setup.py file
+                if self.code_model.options.get("generate-setup-py") or self.code_model.options.get("basic-setup-py"):
+                    # Write the setup file
+                    if self.code_model.options["basic-setup-py"]:
+                        self.write_file(exec_path / Path("setup.py"), general_serializer.serialize_setup_file())
+                else:
+                    self.remove_file(exec_path / Path("setup.py"))
```

Comment:
@swathipil It is not intuitive for the judge logic. It is better like:

```
if basic-setup-py:
  # write setup
elif self.code_model.options.get("generate-setup-py") is False:
  # remove setup.py

```

Reply:
updated!